From 76db36d6e57f4745f5136c4e8195233ef0aa40ce Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <jianzheng.zhou@freescale.com>
Date: Fri, 28 Sep 2012 16:42:00 +0800
Subject: [PATCH 067/602] ENGR00225698 wifi: port atheros wifi sdk

Modify board support file for wifi.
1.change wpa_supplicant to p2p_supplicant which is google's
  default svc name in JB.
2.In JB, HAL wifi change from
    "snprintf(daemon_cmd, PROPERTY_VALUE_MAX, "%s:-i%s -c%s",
              SUPPLICANT_NAME, iface, config_file);
    property_set("ctl.start", daemon_cmd);"
to
    property_set("ctl.start", supplicant_name);
so -i, -c parameters should be added in init.rc.

Change-Id: I00074c111670a8823abfabdb893df7b83a0d01bb
Signed-off-by: Jianzheng Zhou <jianzheng.zhou@freescale.com>
---
 imx6/sabresd/SabreSDBoardConfigComm.mk | 16 +++++++---------
 imx6/sabresd/init.rc                   |  5 +++--
 2 files changed, 10 insertions(+), 11 deletions(-)

diff --git a/imx6/sabresd/SabreSDBoardConfigComm.mk b/imx6/sabresd/SabreSDBoardConfigComm.mk
index a0d06be..0be5db7 100755
--- a/imx6/sabresd/SabreSDBoardConfigComm.mk
+++ b/imx6/sabresd/SabreSDBoardConfigComm.mk
@@ -9,18 +9,16 @@ TARGET_BOOTLOADER_BOARD_NAME := SABRESD
 PRODUCT_MODEL := SABRESD-MX6DQ
 
 # Wifi
-#BOARD_WLAN_VENDOR 			 := ATHEROS
+BOARD_WLAN_VENDOR 			 := ATHEROS
 # for atheros vendor
 ifeq ($(BOARD_WLAN_VENDOR),ATHEROS)
 BOARD_WLAN_DEVICE			 := ar6003
 BOARD_HAS_ATH_WLAN 			 := true
-BOARD_WLAN_ATHEROS_SDK			 := system/wlan/atheros/compat-wireless
-WPA_SUPPLICANT_VERSION			 := VER_0_8_X
-HOSTAPD_VERSION				 := VER_0_8_ATHEROS
+WPA_SUPPLICANT_VERSION			 := VER_0_8_ATHEROS
 WIFI_DRIVER_MODULE_PATH          	 := "/system/lib/modules/ath6kl_sdio.ko"
 WIFI_DRIVER_MODULE_NAME          	 := "ath6kl_sdio"
-WIFI_DRIVER_MODULE_ARG           	 := "suspend_mode=3 ath6kl_p2p=1"
-WIFI_DRIVER_P2P_MODULE_ARG       	 := "suspend_mode=3 ath6kl_p2p=1 debug_mask=0x2413"
+WIFI_DRIVER_MODULE_ARG           	 := "suspend_mode=3 wow_mode=2 ar6k_clock=26000000 ath6kl_p2p=1"
+WIFI_DRIVER_P2P_MODULE_ARG       	 := "suspend_mode=3 wow_mode=2 ar6k_clock=26000000 ath6kl_p2p=1 debug_mask=0x2413"
 WIFI_SDIO_IF_DRIVER_MODULE_PATH  	 := "/system/lib/modules/cfg80211.ko"
 WIFI_SDIO_IF_DRIVER_MODULE_NAME  	 := "cfg80211"
 WIFI_SDIO_IF_DRIVER_MODULE_ARG   	 := ""
@@ -38,9 +36,9 @@ WIFI_DRIVER_MODULE_PATH          	 := "/system/lib/modules/iwlagn.ko"
 WIFI_DRIVER_MODULE_NAME          	 := "iwlagn"
 WIFI_DRIVER_MODULE_PATH			 ?= auto
 endif
-#BOARD_WPA_SUPPLICANT_DRIVER      	 := NL80211
-#BOARD_HOSTAPD_DRIVER             	 := NL80211
-#WIFI_TEST_INTERFACE			 := "sta"
+BOARD_WPA_SUPPLICANT_DRIVER      	 := NL80211
+BOARD_HOSTAPD_DRIVER             	 := NL80211
+WIFI_TEST_INTERFACE			 := "sta"
 
 BOARD_MODEM_VENDOR := AMAZON
 
diff --git a/imx6/sabresd/init.rc b/imx6/sabresd/init.rc
index c59c800..6174a42 100755
--- a/imx6/sabresd/init.rc
+++ b/imx6/sabresd/init.rc
@@ -98,8 +98,9 @@ on boot
     setprop hw.bluetooth 0
     setprop wlan.interface wlan0
 
-service wpa_supplicant /system/bin/wpa_supplicant \
-    -Dnl80211 -puse_p2p_group_interface=0  -e/data/misc/wifi/entropy.bin
+service p2p_supplicant /system/bin/wpa_supplicant \
+    -Dnl80211 -puse_p2p_group_interface=0  -e/data/misc/wifi/entropy.bin \
+    -iwlan0 -c/data/misc/wifi/wpa_supplicant.conf
     class late_start
     disabled
     oneshot
-- 
1.8.0

